src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -384,19 +367,15 @@ protected Date expiration() {
    * @return The info about local repositories.
    * @throws IOException If something went wrong.
    */
-  protected Map<URL, LocalRepositoryInfo> loadLocalRepositoriesInfo() throws IOException {
-    synchronized (LOCAL_REPOSITORIES_INFO_LOCK) {
-      Path path = base.resolve(DEFAULT_LOCAL_REPOSITORIES_INFO_FILE);
-
-      if (!Files.exists(path)) {
-        return new HashMap<>();
-      }
-
-      if (!Files.isRegularFile(path)) {
-        throw new IOException(String.format("Hey! %s is not a file!", path));
-      }
-
-      return MAPPER.readValue(Files.newInputStream(path), LOCAL_REPOSITORIES_TYPE_REF);
+  private static void loadLocalRepositoriesInfo() throws IOException {
+    Path path = DEFAULT_DIRECTORY.resolve(DEFAULT_LOCAL_REPOSITORIES_INFO_FILE);
+    if (!Files.exists(path)) {
+      LOCAL_REPOSITORIES_INFO.clear();
+    } else if (!Files.isRegularFile(path)) {
+      throw new IOException(String.format("Hey! %s is not a file!", path));
+    } else {
+      LOCAL_REPOSITORIES_INFO
+          .putAll(MAPPER.readValue(Files.newInputStream(path), LOCAL_REPOSITORIES_TYPE_REF));
```

Comment:
What is the reason for using if-else block here? If no, please revert it, and leave only the necessary updates.